packages/jest-haste-map/src/index.js

```diff
@@ -672,10 +672,7 @@ class HasteMap extends EventEmitter {
    * Helpers
    */
   _ignore(filePath: Path): boolean {
-    return (
-      this._options.ignorePattern.test(filePath) ||
-      (!this._options.retainAllFiles && this._isNodeModulesDir(filePath))
```

Comment:
Instead maybe we could add `config.testPathIgnorePatterns` to `this._whitelist`?